src/Module/WebClient.php

```diff
@@ -641,6 +640,88 @@ public function goto() {
         }
     }
 
+    /**
+     * This function saves event state into our redis database
+     *
+     * It saves event into redis and returns the identifer
+     */
+    public function saveWebClientState() {
+
+        require_once HV_ROOT_DIR.'/../src/Database/ClientState.php';
+
+        $client_state = new ClientState();
+
+        try {
+
+            $state_key = $client_state->upsert($this->_params['state']);
+
+            $this->_printJSON(json_encode([
+                "status_code" => 200,
+                "status_txt" => "OK",
+                "data" => [
+                    "state_id" => $state_key,
+                ]
+            ]));
+
+        } catch (\Exception $e) {
+
+            http_response_code(500);
+
+            $this->_printJSON(json_encode([
+                "status_code" => 500,
+                "status_txt" => "Server Error",
+                "data" => "",
+            ]));
+        }
+
+
+    }
+
+    /**
+     * This function returns the event state for the given id 
+     */
+    public function getWebClientState() 
+    {
+        require_once HV_ROOT_DIR.'/../src/Database/ClientState.php';
+
+        $client_state = new ClientState();
+
+        try {
+
+            $state = $client_state->find($this->_params['state_id']);
+
+            if(is_null($state)) {
+                http_response_code(404);
+                $this->_printJSON(json_encode([
+                    "status_code" => 404,
+                    "status_txt" => "Not Found",
+                    "data" => "",
+                ]));
+                return;
+            }
+
+            $this->_printJSON(json_encode([
+                "status_code" => 200,
+                "status_txt" => "OK",
+                "data" => [
+                    "state" => $state,
+                ]
+            ]));
+
+        } catch (\Exception $e) {
+
+            http_response_code(500);
+
+            $this->_printJSON(json_encode([
+                "status_code" => 500,
+                "status_txt" => "Server Error",
+                "data" => "",
+            ]));
```

Comment:
I like this way of error handling.

Can you add a function like this?
So strings don't have to be re-typed each time. Less change of typos.
```
make_response($code, $message, $data) {
  return json_encode([
    "status_code" => $code,
    "status_txt" => $message,
    "data" => $data
  ]);
}
```

Reply:
yeah excellent idea! , let me do that